File: 413-arithmetic-slices/413-arithmetic-slices.py
class Solution:
    def numberOfArithmeticSlices(self, nums: List[int]) -> int:
        dp = [0]*len(nums)
        diff,left = 0,0
        for right in range(1,len(nums)):
            if right==1:
                diff = nums[right]-nums[right-1]
            else:
                if diff!=nums[right]-nums[right-1]:
                    diff = nums[right]-nums[right-1]
                    left = right-1
                    dp[right] = dp[right-1]
                else:
                    dp[right] = dp[right-1] + right-left - 1
        return dp[-1]
        

File: 413-arithmetic-slices/413-arithmetic-slices.java
class Solution {
    public int numberOfArithmeticSlices(int[] nums) {
        int[] dp = new int[nums.length];
        int left = 0, diff = 0;
        for(int right=1;right<nums.length;right++){
            if(right==1){
                diff = nums[right] - nums[right-1];
            }else{
                if(diff!=nums[right]-nums[right-1]){
                    diff = nums[right]-nums[right-1];
                    dp[right] = dp[right-1];
                    left = right - 1;
                }else{
                    dp[right] = dp[right-1] + right - left - 1;
                }
            }
        }
        return dp[nums.length-1];
    }
}